modules/transformer-kinesis/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/transformer/metrics/Metrics.scala

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2022 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.snowplow.rdbloader.transformer.metrics
+
+import cats.implicits._
+import cats.{Applicative, Monad}
+import cats.effect.concurrent.Ref
+import cats.effect.{Async, Blocker, ConcurrentEffect, ContextShift, Resource, Sync, Timer}
+
+import fs2.Stream
+
+import scala.concurrent.duration.FiniteDuration
+
+import com.snowplowanalytics.snowplow.rdbloader.common.config.TransformerConfig.MetricsReporters
+
+trait Metrics[F[_]] {
+
+  /** Stream that sends latest metrics via reporter(s) */
+  def report: Stream[F, Unit]
+
+  /** Increment count of successfully transformed events */
+  def goodCount: F[Unit]
+
+  /** Increment count of events that couldn't get transformed */
+  def badCount: F[Unit]
+}
+
+object Metrics {
+
+  val goodCounterName = "good"
+  val badCounterName = "bad"
+  
+  private val defaultPrefix = "snowplow.transformer"
+
+  final case class MetricSnapshot(
+    goodCount: Int,
+    badCount: Int
+  )
+
+  def build[F[_]: ContextShift: ConcurrentEffect: Timer](
+    blocker: Blocker,
+    config: MetricsReporters
+  ): F[Metrics[F]] =
+    config match {
+      case MetricsReporters(None, None) => noop[F].pure[F]
+      case MetricsReporters(statsd, stdout) => impl[F](blocker, statsd, stdout)
+    }
+
+  private def impl[F[_]: ContextShift: ConcurrentEffect: Timer](
+    blocker: Blocker,
+    statsd: Option[MetricsReporters.StatsD],
+    stdout: Option[MetricsReporters.Stdout]
+  ): F[Metrics[F]] =
+    for {
+      refsStatsd <- MetricRefs.init[F]
```

Comment:
Why there are two separate `MetricsRefs` for statsd and stdout? Wouldn't it be possible to have one`MetricRefs` shared by reporters?